### ISSUE TITLE
Livewire loosing $this with nested components in a blade component slot.

### ISSUE THREAD

**Issue by christophheich** [livewire/livewire#2380]
### Description
If a livewire component is used within a blade component within a livewire component there is no way to access a variable anymore, and an exception will be thrown. Using $this when not in object context.

https://github.com/livewire/livewire/discussions/2313#discussioncomment-293525
https://github.com/livewire/livewire/discussions/2313#discussioncomment-285714

### Exact steps to reproduce

Create a livewire component and a nested livewire component as well as blade component with a named or default slot.

Add the blade component into the livewire component and add the nested component into the blade component.

Trying to access a livewire variable (declared / initialized in the parent livewire component) in the blade component should now throw the exception, if the nested component in the blade slot will be removed, it works as expected.

```html
(Livewire Component)
<div>
    <h1>Random</h1>
    (Blade Component)
    <x-random-blade-component>
        (Nested Livewire Component in a blade slot)
        @livewire('random-nested-livewire-component')
    </x-random-blade-component>
</div>
```

### Stripped-down, copy-pastable code snippets

```html
(Livewire Component)
<div>
    <h1>Random</h1>
    (Blade Component)
    <x-random-blade-component>
        (Nested Livewire Component in a blade slot)
        @livewire('random-nested-livewire-component')
    </x-random-blade-component>
</div>
```

**Context**
- Livewire version: 2.3.6
- Laravel version: 8.22.1
- Alpine version: 2.8.0
- Browser: Chrome, Firefox

**Comment by redbastie** [comment]
You can't use `$this` inside of an included view. You have to use `$this` directly inside the parent view.

**Comment by redbastie** [comment]
What you can do for computed properties, is pass them to the include as a variable. For example

    @include('includes.infinite-scroll', ['infiniteScroll' => $this->infiniteScroll])

**Comment by christophheich** [comment]
> What you can do for computed properties, is pass them to the include as a variable. For example
> 
> ```
> @include('includes.infinite-scroll', ['infiniteScroll' => $this->infiniteScroll])
> ```

Okay, i'll try to explain it on another way.

If i am using a livewire component, which has a blade component (https://laravel.com/docs/8.x/blade#components) inside of it, i can call $this in the blade component and it will work just fine. The blade component can access $this from the parent livewire component.

```
(Livewire Component)
<div>
    <h1>Random</h1>
    <x-random-blade-component /> ($this will be called inside the blade component and it just works)
</div>
```

As soon as i add a second (nested) livewire component inside of the blade component (which is inside the livewire component) $this will throw an error, even though i am not accessing it in the nested blade component.

```
(Livewire Component)
<div>
    <h1>Random</h1>
    <x-random-blade-component> (still calling $this only in the blade component an error will be thrown)
        @livewire('random-nested-livewire-component')
    </x-random-blade-component>
</div>
```

@include would not be an option (bad formatting and passing over html bla bla)

if that's causing the issue and is a normal behaviour i guess i need to pass the $this->infiniteScroll to the component using :infiniteScroll="$this->infiniteScroll"

<x-random-blade-component  :infiniteScroll="$this->infiniteScroll">

**Comment by MarioPerini** [comment]
Same Issue here.
Here a example repo: https://github.com/MarioPerini/LivewireChildIssue

**Comment by HelgeSverre** [comment]
I am also seeing similar behavior.

I have a main livewire component, with a nested component inside of it, i also use @include() as part of the root component to include a tab, inside of the included blade file i am referring to $this->someComputedPropertyInRootComponent, and this throws a "Using $this when not in object context." error.

**Comment by joshhanley** [comment]
I have added a failing test for this in PR #2786.

There is also a temporary work around listed in the PR, if anyone is having this issue.

Hope this helps!

**Comment by tanthammar** [comment]
Confirmed: Solved with  #2786. THANK YOU @joshhanley

**Comment by squishythejellyfish** [comment]
👋 Oh Hi! I'm Squishy, the friendly jellyfish that manages Livewire issues.

I see this issue has been closed.

Here in the Livewire repo, we have an "issues can be closed guilt-free and without explanation" policy.

If for ANY reason you think this issue hasn't been resolved, PLEASE feel empowered to re-open it.

Re-opening actually helps us track which issues are a priority.

Reply "REOPEN" to this comment and we'll happily re-open it for you!

(More info on this philosophy here: https://twitter.com/calebporzio/status/1321864801295978497)
